File: zad1&2/Tipper/Tipper/app/src/main/java/com/example/tipper/activities/QuizActivity.java
package com.example.tipper.activities;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;
import com.example.tipper.R;

public class QuizActivity extends AppCompatActivity {

    QuizController controller;
    TextView questionTextView;
    Button answer1btn;
    Button answer2btn;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState); // call superclass onCreate
        setContentView(R.layout.activity_quiz); // inflate the GUI

        questionTextView = (TextView) findViewById(R.id.questionTextView);
        answer1btn = (Button) findViewById(R.id.answer1button);
        answer2btn = (Button) findViewById(R.id.answer2button);

        controller = new QuizController();

        SetUpQuestionAndAnswer();

        answer1btn.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View view){
                controller.Answer(0);
                checkContinuityOfQuiz();
            }
        });

        answer2btn.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View view){
                controller.Answer(1);
                checkContinuityOfQuiz();
            }
        });
    }

    private void checkContinuityOfQuiz() {
        if(controller.isFinished()){
            questionTextView.setText("Udzieliłeś " + controller.getGoodAnswers() + " poprawnych odpowiedzi z " + controller.getQuestionsAndAnswers().size());
            answer1btn.setVisibility(View.INVISIBLE);
            answer2btn.setVisibility(View.INVISIBLE);
        }
        else{
            SetUpQuestionAndAnswer();
        }
    }

    private void SetUpQuestionAndAnswer(){
        questionTextView.setText(controller.GetCurrentQuestion());
        answer1btn.setText(controller.GetCurrentAnswer1());
        answer2btn.setText(controller.GetCurrentAnswer2());
    }

}

File: zad1&2/Tipper/Tipper/app/src/main/java/com/example/tipper/activities/QuizActivity.kt
package com.example.tipper.activities

import android.os.Bundle
import com.example.tipper.R
import android.view.*
import android.widget.*
import androidx.appcompat.app.AppCompatActivity

class QuizActivity : AppCompatActivity() {
    var controller: QuizController? = null
    var questionTextView: TextView? = null
    var answer1btn: Button? = null
    var answer2btn: Button? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState) // call superclass onCreate
        setContentView(R.layout.activity_quiz) // inflate the GUI
        questionTextView = findViewById<View>(R.id.questionTextView) as TextView
        answer1btn = findViewById<View>(R.id.answer1button) as Button
        answer2btn = findViewById<View>(R.id.answer2button) as Button
        controller = QuizController(this.applicationContext)
        SetUpQuestionAndAnswer()
        answer1btn!!.setOnClickListener {
            controller!!.Answer(0)
            checkContinuityOfQuiz()
        }
        answer2btn!!.setOnClickListener {
            controller!!.Answer(1)
            checkContinuityOfQuiz()
        }
    }

    private fun checkContinuityOfQuiz() {
        if (controller!!.isFinished()) {
            questionTextView!!.text = "Udzieliłeś " + controller!!.getGoodAnswers() + " poprawnych odpowiedzi z " + controller!!.getQuestionsAndAnswers()!!.size
            answer1btn!!.visibility = View.INVISIBLE
            answer2btn!!.visibility = View.INVISIBLE
        } else {
            SetUpQuestionAndAnswer()
        }
    }

    private fun SetUpQuestionAndAnswer() {
        questionTextView!!.text = controller!!.GetCurrentQuestion()
        answer1btn!!.text = controller!!.GetCurrentAnswer1()
        answer2btn!!.text = controller!!.GetCurrentAnswer2()
    }
}